### ISSUE TITLE
sometimes nanovg crashes on fonsTextIterInit

### ISSUE THREAD

**Issue by tailangjun** [memononen/nanovg#383]
sometimes nanovg crashes on fonsTextIterInit when I call nvgText(). maybe the stbtt_fontinfo has released?

(fonsTextIterInit,stb_truetype.h,line 2214,col 32 + 1917040)
(fonsTextIterInit,fontstash.h,line 1332,col 2 + 1916912)

**Comment by memononen** [comment]
The line numbers does not make sense. Are you using the latest code?

**Comment by tailangjun** [comment]
I use the latest code of nanovg and stb_truetype, the latest stb_truetype use 'CFF ' table. 
<img width="762" alt="screen shot 2017-06-28 at 12 59 07 pm" src="https://user-images.githubusercontent.com/12316965/27621207-cb6b2bac-5c01-11e7-85b9-eda9c749e584.png">
Order of function calling:nvgText()->fonsTextIterInit()->fons__tt_getPixelHeightScale()->stbtt_ScaleForPixelHeight()

I invoke nvgCreateFontMem() on main thread, and then invoke nvgText() on sub thread. I make sure stbtt_InitFont() has been finished when invoke nvgText(). so I can not understand why crashed on stbtt_ScaleForPixelHeight().

**Comment by memononen** [comment]
NanoVG is not thread safe by any means.

When you get the crash, hoe does things look like in debugger? Does the font look ok on main thread and is the `info` valid at the point where it crashes? Does `nvgDeleteInternal` get called on main thread before you call `nvgText`?

**Comment by tailangjun** [comment]
You are right. I move the create() and destory() invoke logic to the subthread, not crashed anymore. In addition, add reset fs pointer,
`	
if (ctx->fs) {
	fonsDeleteInternal(ctx->fs);
       ctx->fs = NULL;
};
`
